From 8133bfd2165c4c9104207c938e5f6482232981ed Mon Sep 17 00:00:00 2001
From: Pier Luigi Fiorini <pierluigi.fiorini@gmail.com>
Date: Wed, 24 Jul 2013 23:19:27 +0200
Subject: [PATCH] Override modules list and don't always load desktop-shell.so

Let --modules override modules list and load desktop-shell.so as a
fallback if a modules list is not specified neither by passing
--modules nor with weston.ini.

The original patch is there:
http://lists.freedesktop.org/archives/wayland-devel/2013-April/008322.html

It should be implemented in a different way but for now this patch
is enough.
---
 src/compositor.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/compositor.c b/src/compositor.c
index e9e1166..c4829c6 100644
--- a/src/compositor.c
+++ b/src/compositor.c
@@ -3348,9 +3348,7 @@ int main(int argc, char *argv[])
 
 	setenv("WAYLAND_DISPLAY", socket_name, 1);
 
-	if (load_modules(ec, modules, &argc, argv) < 0)
-		goto out;
-	if (load_modules(ec, option_modules, &argc, argv) < 0)
+	if (load_modules(ec, option_modules ? option_modules : modules, &argc, argv, config_file) < 0)
 		goto out;
 
 	for (i = 1; i < argc; i++)
-- 
1.8.1.4

